cmd/okctl/main.go

```diff
@@ -56,6 +56,10 @@ We also use the prometheus-operator for ensuring metrics and logs are
 being captured. Together with slack and slick.`,
 		SilenceUsage: true,
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if cmd.Name() == cobra.ShellCompRequestCmd {
+				return nil
+			}
```

Comment:
Not fantastic, but `__completion` is built right into `cobra.Command` without a `PersistantPreRunE`.

The other solution I can think of is to not have this "init"-style `PersistantPreRunE` here on root, but instead move it out to the subcommands that requires it to work. (Then you'd also remove the empty `PersistantPreRunE` on subcommands such as `version`).

Reply:
I think this approach is good for now, we will eventually only support the declarative approach at which case we will remove the `PersistentPreRunE` since there will only be one location where we need to perform this same initialization.

Reply:
Getting lots of errors on code I haven't touched? And it reformats a bunch of files as well? :grimacing: 

```
/home/eide/go/bin/golangci-lint run
WARN [runner/nolint] Found unknown linters in //nolint directives: funlen gocognit, funlen gocyclo, funlen lll, gomnd funlen lll 
pkg/helm/helm.go:290: Function 'Install' has too many statements (76 > 40) (funlen)
func (h *Helm) Install(kubeConfigPath string, cfg *InstallConfig) (*release.Release, error) {
cmd/okctl/completion.go:8: Function 'buildCompletionCommand' is too long (67 > 60) (funlen)
func buildCompletionCommand(o *okctl.Okctl) *cobra.Command {
cmd/okctl/create.go:130: Function 'buildCreateClusterCommand' is too long (351 > 60) (funlen)
func buildCreateClusterCommand(o *okctl.Okctl) *cobra.Command {
cmd/okctl/createtestcluster.go:75: Function 'buildCreateTestClusterCommand' is too long (281 > 60) (funlen)
func buildCreateTestClusterCommand(o *okctl.Okctl) *cobra.Command {
cmd/okctl/deletetestcluster.go:42: Function 'buildDeleteTestClusterCommand' is too long (146 > 60) (funlen)
func buildDeleteTestClusterCommand(o *okctl.Okctl) *cobra.Command {
cmd/okctl/show.go:38: Function 'buildShowCredentialsCommand' is too long (107 > 60) (funlen)
func buildShowCredentialsCommand(o *okctl.Okctl) *cobra.Command {
cmd/okctl/create.go:465:9: ineffectual assignment to `err` (ineffassign)
			txt, err := commands.GoTemplateToString(commands.CreateClusterEndMsg, msg)
			     ^
cmd/okctl/createtestcluster.go:340:9: ineffectual assignment to `err` (ineffassign)
			txt, err := commands.GoTemplateToString(commands.CreateTestClusterEndMsg, msg)
			     ^
pkg/helm/charts/argocd/argocd.go:45: Function 'NewDefaultValues' is too long (284 > 60) (funlen)
func NewDefaultValues(opts ValuesOpts) *Values {
pkg/helm/charts/argocd/argocd.go:56:19: mnd: Magic number: 999, in <assign> detected (gomnd)
				RunAsGroup:   999,
				              ^
pkg/helm/charts/argocd/argocd.go:57:19: mnd: Magic number: 999, in <assign> detected (gomnd)
				RunAsUser:    999,
				              ^
pkg/helm/charts/argocd/argocd.go:58:19: mnd: Magic number: 999, in <assign> detected (gomnd)
				FsGroup:      999,
				              ^
pkg/helm/charts/argocd/argocd.go:69:19: mnd: Magic number: 8082, in <assign> detected (gomnd)
			ContainerPort: 8082,
			               ^
pkg/helm/charts/argocd/argocd.go:71:26: mnd: Magic number: 3, in <assign> detected (gomnd)
				FailureThreshold:    3,
				                     ^
pkg/helm/charts/argocd/argocd.go:72:26: mnd: Magic number: 10, in <assign> detected (gomnd)
				InitialDelaySeconds: 10,
				                     ^
pkg/helm/charts/argocd/argocd.go:73:26: mnd: Magic number: 10, in <assign> detected (gomnd)
				PeriodSeconds:       10,
				                     ^
pkg/helm/charts/argocd/argocd.go:78:26: mnd: Magic number: 3, in <assign> detected (gomnd)
				FailureThreshold:    3,
				                     ^
pkg/helm/charts/argocd/argocd.go:79:26: mnd: Magic number: 10, in <assign> detected (gomnd)
				InitialDelaySeconds: 10,
				                     ^
pkg/helm/charts/argocd/argocd.go:85:15: mnd: Magic number: 8082, in <assign> detected (gomnd)
				Port:     8082,
				          ^
pkg/helm/charts/argocd/argocd.go:130:23: mnd: Magic number: 5556, in <assign> detected (gomnd)
			ContainerPortHTTP: 5556,
			                   ^
pkg/helm/charts/argocd/argocd.go:131:23: mnd: Magic number: 5556, in <assign> detected (gomnd)
			ServicePortHTTP:   5556,
			                   ^
pkg/helm/charts/argocd/argocd.go:132:23: mnd: Magic number: 5557, in <assign> detected (gomnd)
			ContainerPortGrpc: 5557,
			                   ^
pkg/helm/charts/argocd/argocd.go:133:23: mnd: Magic number: 5557, in <assign> detected (gomnd)
			ServicePortGrpc:   5557,
			                   ^
pkg/helm/charts/argocd/argocd.go:148:19: mnd: Magic number: 6379, in <assign> detected (gomnd)
			ContainerPort: 6379,
			               ^
pkg/helm/charts/argocd/argocd.go:149:19: mnd: Magic number: 6379, in <assign> detected (gomnd)
			ServicePort:   6379,
			               ^
pkg/helm/charts/argocd/argocd.go:157:19: mnd: Magic number: 1000, in <assign> detected (gomnd)
				RunAsGroup:   1000,
				              ^
pkg/helm/charts/argocd/argocd.go:158:19: mnd: Magic number: 1000, in <assign> detected (gomnd)
				RunAsUser:    1000,
				              ^
pkg/helm/charts/argocd/argocd.go:159:19: mnd: Magic number: 1000, in <assign> detected (gomnd)
				FsGroup:      1000,
				              ^
pkg/helm/charts/argocd/argocd.go:178:40: mnd: Magic number: 5, in <assign> detected (gomnd)
				MaxReplicas:                       5,
				                                   ^
pkg/helm/charts/argocd/argocd.go:179:40: mnd: Magic number: 50, in <assign> detected (gomnd)
				TargetCPUUtilizationPercentage:    50,
				                                   ^
pkg/helm/charts/argocd/argocd.go:180:40: mnd: Magic number: 50, in <assign> detected (gomnd)
				TargetMemoryUtilizationPercentage: 50,
				                                   ^
pkg/helm/charts/argocd/argocd.go:184:19: mnd: Magic number: 8080, in <assign> detected (gomnd)
			ContainerPort: 8080,
			               ^
pkg/helm/charts/argocd/argocd.go:186:26: mnd: Magic number: 3, in <assign> detected (gomnd)
				FailureThreshold:    3,
				                     ^
pkg/helm/charts/argocd/argocd.go:214:31: mnd: Magic number: 80, in <assign> detected (gomnd)
				ServicePortHTTP:          80,
				                          ^
pkg/helm/charts/argocd/argocd.go:215:31: mnd: Magic number: 443, in <assign> detected (gomnd)
				ServicePortHTTPS:         443,
				                          ^
pkg/helm/charts/argocd/argocd.go:284:40: mnd: Magic number: 5, in <assign> detected (gomnd)
				MaxReplicas:                       5,
				                                   ^
pkg/helm/charts/argocd/argocd.go:285:40: mnd: Magic number: 50, in <assign> detected (gomnd)
				TargetCPUUtilizationPercentage:    50,
				                                   ^
pkg/helm/charts/argocd/argocd.go:289:19: mnd: Magic number: 8081, in <assign> detected (gomnd)
			ContainerPort: 8081,
			               ^
pkg/helm/charts/argocd/argocd.go:315:15: mnd: Magic number: 8081, in <assign> detected (gomnd)
				Port:     8081,
				          ^
pkg/ask/ask_test.go:92: Function 'TestAskSelectInfrastructureRepository' is too long (75 > 60) (funlen)
func TestAskSelectInfrastructureRepository(t *testing.T) {
pkg/ask/ask_test.go:171: Function 'TestAskSelectTeam' is too long (70 > 60) (funlen)
func TestAskSelectTeam(t *testing.T) {
pkg/ask/ask_test.go:245: Function 'TestAskCreateOauthApp' is too long (83 > 60) (funlen)
func TestAskCreateOauthApp(t *testing.T) {
pkg/ask/ask_test.go:117: line is 180 characters (lll)
				_, _ = c.ExpectString("? Select infrastructure as code repository:  [Use arrows to move, type to filter, ? for more help]\r\n  oslokommune/something\r\n> oslokommune/else\r\n")
pkg/ask/ask_test.go:192: line is 180 characters (lll)
				_, _ = c.ExpectString("? Select team that is authorised to access the Argo CD UI:  [Use arrows to move, type to filter, ? for more help]\r\n> Super team\r\n  Awesome team\r\n")
pkg/client/core/report/console/console.go:30:1: cognitive complexity 33 of func `(*Console).Report` is high (> 30) (gocognit)
func (c *Console) Report(actions []store.Action, component, description string) error {
^
make: *** [Makefile:133: lint] Error 1

```

Reply:
What is the output from: `golangci-lint version`?

Reply:
To satisfy I think it will suffice to add `// nolint: funlen` to `cmd/okctl/completion.go:8: Function 'buildCompletionCommand' is too long (67 > 60) (funlen)`

Reply:
```
golangci-lint has version v1.36.1-0.20210208193152-3ef13a8028c1 built from (unknown, mod sum: "h1:6j8+FDOshmSGug3G7jjCT+2DwefGB4nVpHuEaB7V080=") on (unknown)
```

Reply:
```
➜  okctl git:(add-nightly-test) golangci-lint version
golangci-lint has version 1.32.2 built from 52d26a3 on 2020-11-03T01:15:38Z
```

Reply:
We use a specific version of `golangci-lint`, if you run from the root of the okctl repo it should work:

```bash
rm $(go env GOBIN)/golangci-lint && make check
```

Reply:
I forgot to say that `make check` will fetch the correct version of `golangci-lint`